lib/compress/zstd_compress.c

```diff
@@ -1098,6 +1101,14 @@ size_t ZSTD_CCtx_setParametersUsingCCtxParams(
     return 0;
 }
 
+size_t ZSTD_CCtx_setCParams(ZSTD_CCtx* cctx, ZSTD_compressionParameters cparams)
+{
+    FORWARD_IF_ERROR(ZSTD_checkCParams(cparams), "");
+    assert(cctx != NULL);
```

Comment:
Yes, some are,
the direct equivalent to reproduce would be : https://github.com/facebook/zstd/blob/dev/lib/compress/zstd_compress.c#L659

And, if the same logic is present multiple times, it would be better to factor it into a common function.